File: TimeTrackerPro/androidApp/src/main/java/com/appsdeviser/timetrackerpro/android/ui/core/error/dialog/DialogState.kt
package com.appsdeviser.timetrackerpro.android.ui.core.error.dialog

data class DialogState(
    val title: String = "No Title",
    val description: String = "No Description",
    val positiveButtonText: String = "OK",
    val negativeButtonText: String? = null,
    val isCancelable: Boolean = true,
    val priority: DialogPriority = DialogPriority.UNKNOWN
)

sealed class DialogPriority(val priority: Int) {
    data object HIGH: DialogPriority(1)
    data object MEDIUM: DialogPriority(2)
    data object LOW: DialogPriority(3)
    data object UNKNOWN: DialogPriority(1000)
}

File: TimeTrackerPro/iosApp/iosApp/core/dialog/DialogState.swift
//
//  DialogState.swift
//  iosApp
//
//  Created by AndroidLab on 2024-04-05.
//  Copyright © 2024 orgName. All rights reserved.
//

enum DialogPriority: Int {
    case HIGH = 1
    case MEDIUM = 2
    case LOW = 3
    case UNKNOWN = 1000
}

// Struct representing dialog state
struct DialogState {
    var title: String
    var description: String
    var positiveButtonText: String
    var negativeButtonText: String?
    var isCancelable: Bool
    var priority: DialogPriority
    
    init(title: String = "No Title",
         description: String = "No Description",
         positiveButtonText: String = "OK",
         negativeButtonText: String? = nil,
         isCancelable: Bool = true,
         priority: DialogPriority = .UNKNOWN) {
        self.title = title
        self.description = description
        self.positiveButtonText = positiveButtonText
        self.negativeButtonText = negativeButtonText
        self.isCancelable = isCancelable
        self.priority = priority
    }
}
